sklearn/utils/validation.py

```diff
@@ -402,13 +402,20 @@ def check_array(array, accept_sparse=False, dtype="numeric", order=None,
         array = np.array(array, dtype=dtype, order=order, copy=copy)
 
         if ensure_2d:
+            # If input is scalar raise error
+            if array.ndim == 0:
+                raise ValueError(
+                    "Expected 2D array, got scalar array instead:\narray={}.\n"
+                    "Reshape your data either using array.reshape(-1, 1) if "
```

Comment:
Does this apply to a scalar array?

Reply:
I would say the advice does not apply to scalar arrays.

Reply:
@liamge do you intend to address this?

Reply:
Can reshape(1,1)

Reply:
``np.array(10).reshape(-1, 1)`` actually works (np 1.13.1).

Reply:
Maybe you can rewrite the code to avoid the copy and paste look and feel:


```py
if array.ndim < 2:
    array_type = 'scalar' if array.ndim == 0 else '1D'
    raise ValueError('...got {} array instead ... array={} ...'.format(array_type, array))
```

Reply:
Very sorry about the radio silence, would you want me to add an additional commit with the avoidance of the copy/paste look?

Reply:
Well @amueller reckons this works, do let's just key it.